code/modules/mob/living/simple_animal/eldritch_demons.dm

```diff
@@ -0,0 +1,275 @@
+
+/mob/living/simple_animal/hostile/eldritch/raw_prophet/Login()
+	. = ..()
+	client?.view_size.setTo(11)
+
+/mob/living/simple_animal/hostile/eldritch/armsy
+	name = "Terror of the night"
+	real_name = "Armsy"
+	desc = "Abomination made from severed limbs."
+	icon_state = "armsy_start"
+	icon_living = "armsy_start"
+	maxHealth = 200
+	health = 200
+	melee_damage_lower = 10
+	melee_damage_upper = 15
+	move_resist = MOVE_FORCE_OVERPOWERING+1
+	///Previous segment in the chain
+	var/mob/living/simple_animal/hostile/eldritch/armsy/back
+	///Next segment in the chain
+	var/mob/living/simple_animal/hostile/eldritch/armsy/front
+	///Your old location
+	var/oldloc
+	///Allow / disallow pulling
+	var/allow_pulling = FALSE
+	///How many arms do we have to eat to expand?
+	var/stacks_to_grow = 5
+	///Currently eaten arms
+	var/current_stacks = 0
+
+/mob/living/simple_animal/hostile/eldritch/armsy/New(spawn_more = TRUE,len = 6)
+	. = ..()
+	if(len < 3)
+		stack_trace("Eldritch Armsy created with invalid len ([len]). Reverting to 3.")
+		len = 3 //code breaks below 3, let's just not allow it.
+	oldloc = loc
+	if(!spawn_more)
+		return
+	allow_pulling = TRUE
+	///next link
+	var/mob/living/simple_animal/hostile/eldritch/armsy/next
+	///previous link
+	var/mob/living/simple_animal/hostile/eldritch/armsy/prev
+	///current link
+	var/mob/living/simple_animal/hostile/eldritch/armsy/current
+	for(var/i in 0 to len)
+		prev = current
+		//i tried using switch, but byond is really fucky and it didnt work as intended. Im sorry
+		if(i == 0)
+			current = new type(drop_location(),spawn_more = FALSE)
+			current.icon_state = "armsy_mid"
+			current.icon_living = "armsy_mid"
+			current.front = src
+			current.AIStatus = AI_OFF
+			back = current
+		else if(i < len)
+			current = new type(drop_location(),spawn_more = FALSE)
+			prev.back = current
+			prev.icon_state = "armsy_mid"
+			prev.icon_living = "armsy_mid"
+			prev.front = next
+			prev.AIStatus = AI_OFF
+		else
+			prev.icon_state = "armsy_end"
+			prev.icon_living = "armsy_end"
+			prev.front = next
+			prev.AIStatus = AI_OFF
+		next = prev
+
+/mob/living/simple_animal/hostile/eldritch/armsy/can_be_pulled()
+	return FALSE
+
+/mob/living/simple_animal/hostile/eldritch/armsy/Moved()
+	. = ..()
+	gib_trail()
+	if(back && back.loc != oldloc)
+		back.Move(oldloc)
+	if(front && loc != front.oldloc)
+		forceMove(front.oldloc) // self fixing properties
+	oldloc = loc
+
+/mob/living/simple_animal/hostile/eldritch/armsy/proc/gib_trail()
+	if(front) // head makes gibs
+		return
+	var/chosen_decal = pick(typesof(/obj/effect/decal/cleanable/blood/gibs))
+	var/obj/effect/decal/cleanable/blood/gibs/decal = new chosen_decal(drop_location())
+	decal.setDir(dir)
+
+/mob/living/simple_animal/hostile/eldritch/armsy/Destroy()
+	if(front)
+		front.icon_state = "armsy_end"
+		front.icon_living = "armsy_end"
+	if(back)
+		QDEL_NULL(back) // chain destruction baby
+	return ..()
```

Comment:
This mob being destroyed is the `front.back`. That variable should be nulled, no?

Reply:
oh yeah it should be nulled, you are right